ffi/firewood.go

```diff
@@ -222,5 +222,17 @@ func (db *Database) Root() ([]byte, error) {
 
 // Revision returns a historical revision of the database.
 func (db *Database) Revision(root []byte) (*Revision, error) {
-	return newRevision(db.handle, root)
+	if root == nil || len(root) != RootLength {
+		return nil, errInvalidRootLength
+	}
+
+	// Attempt to get any value from the root.
+	// This will verify that the root is valid and accessible.
+	// If the root is not valid, this will return an error.
+	_, err := db.GetFromRoot(root, []byte{})
+	if err != nil {
+		return nil, err
```

Comment:
pre-existing bug, but this is racy. The revision might expire right after this check.

Reply:
Yeah, I realized that earlier. I plan to fix that when I get around to #1008